File: Java/1137. N-th Tribonacci Number.java
class Solution {
    public int tribonacci(int n) {
        int fib[] = new int[38];
        fib[0] = 0; fib[1] = 1; fib[2] = 1;

        for(int i = 3; i <= n; i++) fib[i] = fib[i - 1] + fib[i - 2] + fib[i - 3];

        return fib[n];
    }
}

File: C++/1137. N-th Tribonacci Number.cpp
class Solution {
public:
    int tribonacci(int n) {
        ios_base::sync_with_stdio(false);
        cin.tie(NULL);
        cout.tie(NULL);

        int t[n + 3];
        t[0] = 0; t[1] = 1; t[2] = 1;

        for (int i = 3; i <= n; i++) t[i] = t[i - 1] + t[i - 2] + t[i - 3];

        return t[n];
    }
};
